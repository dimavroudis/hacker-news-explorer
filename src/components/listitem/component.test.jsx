import { render } from "@testing-library/react";
import Listitem from ".";

const ITEMS = [
  {
    title: "React",
    url: "https://reactjs.org/",
    points: 100,
    num_comments: 10,
    author: "Jordan Walke",
    objectID: "0",
  },
  {
    story_title: "Redux",
    story_url: "https://redux.js.org/",
    points: null,
    author: "Dan Abramov",
    objectID: "1",
  },
];

describe("ListItem", () => {
  it("should render correctly with default result", () => {
    const { container } = render(<Listitem item={ITEMS[0]} />);
    expect(container).toMatchSnapshot();
  });
  it("should render correctly with story result", () => {
    const { container } = render(<Listitem item={ITEMS[1]} />);
    expect(container).toMatchSnapshot();
  });
});
