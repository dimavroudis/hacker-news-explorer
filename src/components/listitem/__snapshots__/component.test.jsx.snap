// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListItem > should render correctly with default result 1`] = `
<div>
  <div
    class="_container_8ec90d"
    role="listitem"
  >
    <a
      class="_content_8ec90d"
      href="https://reactjs.org/"
      target="_blank"
    >
      <h3
        class="_title_8ec90d"
      >
        React
      </h3>
      <p
        class="_meta_8ec90d"
      >
        <span>
          100 points
        </span>
        <span
          class="_seperator_8ec90d"
        />
        <span>
           by Jordan Walke
        </span>
        <span
          class="_seperator_8ec90d"
        />
        <span>
          10 comments
        </span>
      </p>
    </a>
  </div>
</div>
`;

exports[`ListItem > should render correctly with story result 1`] = `
<div>
  <div
    class="_container_8ec90d"
    role="listitem"
  >
    <a
      class="_content_8ec90d"
      href="https://redux.js.org/"
      target="_blank"
    >
      <h3
        class="_title_8ec90d"
      >
        Redux
      </h3>
      <p
        class="_meta_8ec90d"
      >
        <span>
          No points
        </span>
        <span
          class="_seperator_8ec90d"
        />
        <span>
           by Dan Abramov
        </span>
        <span
          class="_seperator_8ec90d"
        />
        <span>
          No comments
        </span>
      </p>
    </a>
  </div>
</div>
`;
